ompi/mpi/c/init_thread.c

```diff
@@ -47,28 +47,36 @@ static const char FUNC_NAME[] = "MPI_Init_thread";
 int MPI_Init_thread(int *argc, char ***argv, int required,
                     int *provided)
 {
-    int err;
+    int err, safe_required = MPI_THREAD_SERIALIZED;
 
     ompi_hook_base_mpi_init_thread_top(argc, argv, required, provided);
 
-    if ( MPI_PARAM_CHECK ) {
-        if (required < MPI_THREAD_SINGLE || required > MPI_THREAD_MULTIPLE) {
-            ompi_mpi_errors_are_fatal_comm_handler(NULL, NULL, FUNC_NAME);
-        }
+    /* Detect an incorrect thread support level, but dont report until we have the minimum
+     * infrastructure setup.
+     */
+    if( (MPI_THREAD_SINGLE == required) && (MPI_THREAD_SERIALIZED == required) &&
```

Comment:
i don't think I understand this logic.  shouldn't it be ```||```

Reply:
You're absolutely right, || shall be.